### DISCUSSION TITLE
GitHub assumes .frag files are JavaScript instead of GLSL

### DISCUSSION THREAD

**Discussion by danielhjacobs** [community/community#126603] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

I've never heard of a JavaScript file with the .frag extension, but https://github.com/search?q=repo%3Aruffle-rs%2Fruffle++language%3AJavaScript&type=code shows the .frag files under a search for JavaScript files while only .vert files are shown under a search for GLSL: https://github.com/search?q=repo%3Aruffle-rs%2Fruffle++language%3AGLSL&type=code.

**Comment by LiteBrite82** [comment]
Hi @danielhjacobs 

You are talking about this, right?
![Screenshot 2024-05-31 at 10 58 59 AM](https://github.com/community/community/assets/163985914/9ba573ba-d50e-47e3-ac41-42e19d0162d2)

Yeah, that doesn't seem right. Good find!

**Reply by danielhjacobs** [reply]
Yeah, the .frag files. I suppose a more accurate link to exemplify this would have been https://github.com/search?q=repo%3Aruffle-rs%2Fruffle++language%3AJavaScript+path%3A.frag&type=code, which should have no results, and https://github.com/search?q=repo%3Aruffle-rs%2Fruffle++language%3AGLSL+path%3A.frag&type=code, which should have results.

**Reply by LiteBrite82** [reply]
Though there are .frag files that are listed correctly. When searching for .frag files filtering by language `path:*.frag language:GLSL` the results meet expectations: 
![Screenshot 2024-05-31 at 11 08 24 AM](https://github.com/community/community/assets/163985914/289f20c1-5ee9-4d45-916b-23869efebe2e)

However, the same search filtered for javascript `path:*.frag language:JavaScript `, returns incorrect results:
![Screenshot 2024-05-31 at 11 13 54 AM](https://github.com/community/community/assets/163985914/dac4c64a-b833-4d90-870b-f40a677e131f)

**Reply by LiteBrite82** [reply]
@danielhjacobs you beat me to it! your kung-fu is strong 🥷 😄

**Reply by LiteBrite82** [reply]
Wow, didn't catch that only .7% of the search results accurately classify .frag as GLSL. 1.3k / 178k

**Reply by danielhjacobs** [reply]
Yeah, in that specific repo from my search all the .frag files show up as JavaScript and not GLSL but I can see different results in the other repos. I don't know if it has to do with the exact content of the files or what, as I can't find anything consistent between the correctly vs incorrectly identified files at a glance.

**Reply by LiteBrite82** [reply, marked as answer]
The engineers say thanks, Ive added the bug report to their backlog.

**Reply by abc123me** [reply]
One concern I have, I'm not using GLSL or Javascript but buildroot linux kernel fragment files, if you name a fragment file, say rpi0w-wifi-linux.frag, it shows up as Javascript (or potentially GLSL). These fragment files are used by buildroot in combination with a defconfig provided by Linux for your architecture to configure Linux. As for what language these should show as, I guess Kconfig would be appropriate.

My suggestion would be a way where we can associate a file suffix to a specific language in our repository's settings. I have absolutely no clue how that would be implemented, but I can see it as a concern for generically named files that end in `.frag` symbolizing it's a fragment of some type (whether it be a fragment shader, javascript frag, or buildroot configuration fragment) really should be down to the user to decide.

Here's two examples:
https://github.com/abc123me/jlbsp/blob/master/board/common/linux/headless.frag
https://github.com/abc123me/jlbsp/blob/master/board/common/linux/brcmfmac-wifi.frag
